src/filesystem/fs_misc.cpp

```diff
@@ -0,0 +1,965 @@
+    }
+    else
+        return -1;
+}
+
+const char *fs_get_source_dir_string(const fsc_file_t *file)
+{
+    int id = fs_get_source_dir_id(file);
+    if (id >= 0 && id < FS_MAX_SOURCEDIRS && fs_sourcedirs[id].active)
+        return fs_sourcedirs[id].name;
+    return "unknown";
+}
+
+int fs_get_mod_dir_state(const char *mod_dir)
+{
+    // Returns 3 for current mod, 2 for basemod, 1 for basegame, 0 for inactive mod
+    if (*current_mod_dir && !Q_stricmp(mod_dir, current_mod_dir))
+        return 3;
+    else if (!Q_stricmp(mod_dir, "basemod"))
+        return 2;
+    else if (!Q_stricmp(mod_dir, fs_basegame->string))
+        return 1;
+    return 0;
+}
+
+void fs_file_to_stream(const fsc_file_t *file, fsc_stream_t *stream, qboolean include_source_dir, qboolean include_mod,
+    qboolean include_pk3_origin, qboolean include_size)
+{
+    if (include_source_dir)
+    {
+        fsc_stream_append_string(stream, fs_get_source_dir_string(file));
+        fsc_stream_append_string(stream, "/");
+    }
+    fsc_file_to_stream(file, stream, &fs, include_mod, include_pk3_origin);
+
+    if (include_size)
+    {
+        char buffer[20];
+        Com_sprintf(buffer, sizeof(buffer), " (%i)", file->filesize);
+        fsc_stream_append_string(stream, buffer);
+    }
+}
+
+void fs_file_to_buffer(const fsc_file_t *file, char *buffer, unsigned int buffer_size, qboolean include_source_dir,
+    qboolean include_mod, qboolean include_pk3_origin, qboolean include_size)
+{
+    fsc_stream_t stream = {buffer, 0, buffer_size};
+    fs_file_to_stream(file, &stream, include_source_dir, include_mod, include_pk3_origin, include_size);
+}
+
+void fs_print_file_location(const fsc_file_t *file)
+{
+    char name_buffer[FS_FILE_BUFFER_SIZE];
+    char source_buffer[FS_FILE_BUFFER_SIZE];
+    fs_file_to_buffer(file, name_buffer, sizeof(name_buffer), qfalse, qfalse, qfalse, qfalse);
+    if (file->sourcetype == FSC_SOURCETYPE_PK3)
+    {
+        fs_file_to_buffer((const fsc_file_t *)STACKPTR(((fsc_file_frompk3_t *)file)->source_pk3), source_buffer,
+            sizeof(source_buffer), qtrue, qtrue, qfalse, qfalse);
+        Com_Printf("File %s found in %s\n", name_buffer, source_buffer);
+    }
+    else if (file->sourcetype == FSC_SOURCETYPE_DIRECT)
+    {
+        fs_file_to_buffer(file, source_buffer, sizeof(source_buffer), qtrue, qtrue, qfalse, qfalse);
+        Com_Printf("File %s found at %s\n", name_buffer, source_buffer);
+    }
+    else
+        Com_Printf("File %s has unknown sourcetype\n", name_buffer);
+}
+
+/* ******************************************************************************** */
+// File disabled check
+/* ******************************************************************************** */
+
+static qboolean inactive_mod_file_disabled(const fsc_file_t *file, int level)
+{
+    // Check if a file is disabled by inactive mod settings
+    if (level < 2)
```

Comment:
Invert checks like this; `if ( level >= 2 ) return false;` keep the indentation of the rest of your code close to one level deep as possible.